MSYS2 MINGW-packages mingw-w64-readline/0003-no-winsize.patch, with its hunks refreshed for
readline 8.3.

diff -ruN a/rlwinsize.h b/rlwinsize.h
--- a/rlwinsize.h
+++ b/rlwinsize.h
@@ -55,6 +55,11 @@
 #  define tcflow(fd, action)	ioctl(fd, TCXONC, action)
 #endif
 
+#if defined (__MINGW32__)
+// On Windows, define dummy struct with no fields.
+struct winsize {};
+#endif
+
 extern int _rl_tcgetwinsize (int, struct winsize *);
 extern void _rl_tcsetwinsize (int, struct winsize *);
 
